### ISSUE TITLE
Create a middleware for ring-cors

### ISSUE THREAD

**Issue by stoyle** [muhuk/aleph-middleware#3]
Hi.

I've gotten a PR through ring-cors, to make it easier to write aleph type middleware, since it both handles the request and response. With the new change we are able to do:

```clj
(defn async-response-handler [request access-control response]
  (d/chain' response #(cors/add-access-control request access-control %)))

(defn wrap-cors [handler & access-control]
  (let [access-control (normalize-config access-control)]
    (fn [request]
      (cors/handle-cors handler request access-control async-response-handler))))
```

ring-cors is not a "core" middleware, but I think one that most creating api's would use. The project has no other dependencies than ring and clojure. Is this something you could consider adding support for in aleph-middleware? I guess ring-cors could be an optional dependency.

**Comment by muhuk** [comment]
If we can make it an optional dependency, I am actually ok with it. (BTW [here is an example](https://github.com/muhuk/ring-transit-middleware/blob/master/src/ring/middleware/transit.clj#L48))

**Comment by muhuk** [comment]
I have released `0.2.0` just now.

https://clojars.org/aleph-middleware/versions/0.2.0

Thanks for the contribution!
